pkg/controller/core.oam.dev/v1alpha2/application/revision.go

```diff
@@ -118,12 +118,15 @@ func (h *AppHandler) PrepareCurrentAppRevision(ctx context.Context, af *appfile.
 		return err
 	}
 
-	h.isNewRevision = h.currentAppRevIsNew()
-	if h.isNewRevision {
+	var needGenerateRevision bool
+	h.isNewRevision, needGenerateRevision, err = h.currentAppRevIsNew(ctx)
+	if err != nil {
+		return err
+	}
+	if h.isNewRevision && needGenerateRevision {
 		h.currentAppRev.Name, _ = utils.GetAppNextRevision(h.app)
-	} else {
-		h.currentAppRev = h.latestAppRev.DeepCopy()
```

Comment:
why delete this line?

Reply:
I think we should point the currentAppRev to the existing revision name if exists

Reply:
Move this logic  to function `currentAppRevIsNew()`